### DISCUSSION TITLE
RAD does not boot after adding content folders/files?

### DISCUSSION THREAD

**Discussion by funkychimp** [frntc/RAD#23] [Q&A]
Hello,

My RAD unit boots fine with just the firmware files in the SD Card. However, when I try and add folders and subfolders to the say the RAD_PRG folder I start to get issues.

On adding a few folders it appears to work fine but then after a certain number of file/folders the RAd then refuses to boot. I remove the files/folders and it boots again.

Is there any kind of restrictions with regards to:
- Folder naming conventions
- Number of folders/subfolders allowed
- File name and lengh

Etc.

I have completed dozens of tests adding and removing file/folders but can't seem to find a specific reason as to why its doind this. I have also used multiple SD cards so I know its not the media.

When it it does boot with files on. The .PRG file appear to load without issue. I have ran a few Nuvies also. I get glitches with the Nuvies but am yet to mess with the timings. I am using a C64c PAL unit

Any help would be most appreciated.

Kind regards

Paul

**Comment by frntc** [comment]
There are some (artificial) limitations on the maximum number of files -- how many did you put on your SD card?

Nuvies and glitches: that's probably the timings setting. Some slight adjustments of the last 3 values should do the job (you could start with the last one and see if that already helps)

**Reply by funkychimp** [reply]
Hi frntc,

I first started by puting a full rom set on the SD card. So all .prg files folders A through to Z. I had to split some of the folders down so for example I had folders called C-Ca, Cb-Cj, Cl-Co etc. This was to make sure there were less than 256 files in each folder. I know sometimes in IT this number pops up as a limit here and there. So in total there were 59 subfolders in a folded called 'PRG' which was sat in RAD_PRG. So if we say on average there were 200 files in each folder then maybe in total there were 12,000 approx files.

I removed all the folders and added them in one by one. Sometimes I could get as many as 10 folders on the SD card before it refused to boot. I could not work out a pattern though hence my question. 

Some of the indivudal file names could be quite long also such as 'Captain America in the Doom Tube of Dr. Megalomann (1987)(Go!)[cr 2000 AD].prg'. Just a random example here.

Any thoughts?

Thanks

Paul

**Reply by frntc** [reply]
Since I wanted to keep the code simple, the max. path length is 1024 characters, filenames 256, and it's (also artificially) limited to 1024 files atm. Nothing that couldn't be changed (but I also didn't expect someone putting 12k files on the SD card ;-))

**Reply by funkychimp** [reply]
Hi, 

Right so its the 1024 file limit that is causing my issue. I get your point re the 12,000+ games :). If the main purpuse of this is the play REU files (which it is) then I suppose the 1024 limit is suffuent. I have this plugged into one of my 64s and just wanted to leave it plugged in and have access to the C64 library (all of it). Then I would not need to add other devices such as a SD solution or Pi1541 etc. I am using your Sidekick64 in another 64 and do not have the issue. I just assumed the RAD would act the same. 

I could make a curated list of list of C64 gmes etc. I just own about every popular flash cart ever made and am used to just dumping complete libraries/Rom sets on the SD.

Regards

Paul

**Reply by DoubleA-WTH** [reply]
Hi,

found a MAX_DIR_ENTRIES 16384 def in dirscan.h. 16384 * 4 would sound even nicer :-). 
Cause i also tend to abuse the RAD as a fantastic .prg starter with loads of ram. It's surprising (for me) how many .prg do not run with the REU enabled ...

And found "SD:RAD_PRG", "SD:REU", "SD:GEORAM" as a little lonely BROWSER_NUM_CATEGORIES. 
They could have a friend, like "SD:PRG". For more structure on the SD-card :-).

BTW: Thanks for this great project. Hardware design (including the case) is absolutely fantastic!

Cheers Andreas

**Reply by frntc** [reply]
Hi, 

I actually chose RAD_PRG to use the same SD card for Sidekick64 and RAD but have different .PRGs for each device. I can never imagine that someone copies more than 16k programs/cartridges on the SD card :-)

Can you give some examples which .PRG do not run with REU enabled?

**Reply by DoubleA-WTH** [reply]
Well, the internet is a vast land and .prg are all over there :-)

E.g. Bruce Lee - Return of Fury does start, but then freezes the machine (at least it looks like that, REU at 16M)

**Comment by frntc** [comment]
... had a quick look. For some weird reason it writes to $df00 + x where the REU registers are (while IO range it visible, $01=$35). Of course it can happen that it triggers DMA transfers or whatever. It's also reproduceable in Vice (so no RAD issue, just badly done in the PRG).

**Reply by DoubleA-WTH** [reply]
Yeah, thanks for the research. 
Was pretty sure myself that the REU couldn't be the problem trigger in any case.

BTW: Things i could not test were .crt files. Would like to do that :-) 
And there are looooooaaaaads of .crts out there, too :-)
